File: GanAndroidApp/app/src/main/java/com/xattacker/gan/ServiceFactory.java
package com.xattacker.gan;

import com.xattacker.gan.service.AccountService;
import com.xattacker.gan.service.MsgService;
import com.xattacker.gan.service.SystemService;

//create all service class that only instantiate in the package
final class ServiceFactory
{
	private ServiceFactory()
	{
	}
}

final class InnerAccountService extends AccountService
{
	InnerAccountService(GanClient aAgent, AccountServiceListener aListener)
	{
		super(aAgent, aListener);
	}
}

final class InnerSmsService extends MsgService
{
	InnerSmsService(GanClient aAgent)
	{
		super(aAgent);
	}
}

final class InnerSystemService extends SystemService
{
	InnerSystemService(GanClient aAgent)
	{
		super(aAgent);
	}
}

File: GanAndroidApp/GanClient/src/main/java/com/xattacker/gan/ServiceFactory.kt
package com.xattacker.gan

import com.xattacker.gan.service.AccountService
import com.xattacker.gan.service.MsgService
import com.xattacker.gan.service.SystemService


internal class InnerAccountService(aAgent: GanClient?, aListener: AccountServiceListener?) : AccountService(aAgent, aListener)
internal class InnerMsgService(aAgent: GanClient?) : MsgService(aAgent)
internal class InnerSystemService(aAgent: GanClient?) : SystemService(aAgent)